test/unit/teacher_set_test.rb

```diff
@@ -182,13 +193,293 @@ class TeacherSetTest < ActiveSupport::TestCase
     end
   end
 
+  describe 'create_or_update_teacherset ' do
+    it 'test create or update teacherset' do
+      ts_items_info = {bibs_resp: SIERRA_USER, total_count: 1, available_count: 1, availability_string: 'available'}
+      TeacherSet.instance_variable_set(:@req_body, SIERRA_USER["data"][0])
```

Comment:
So I think this is setting @req_body on the Class! You probably want to put this on line 202 and call it on `teacher_set`